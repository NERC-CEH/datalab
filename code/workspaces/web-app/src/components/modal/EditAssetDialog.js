import React from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import assetLabel from '../common/form/assetLabel';

const useStyles = makeStyles(() => ({
  root: {
    marginTop: 0,
  },
}));

const EditAssetDialog = ({ onSubmit, onCancel, asset, formComponent: Form }) => {
  const classes = useStyles();
  return (
  <Dialog open={true} maxWidth="md">
    <DialogTitle>Edit Asset</DialogTitle>
    <DialogContent>
      <Typography variant="h5" className={classes.root}>{assetLabel(asset)}</Typography>
      <Form onSubmit={onSubmit} onCancel={onCancel} initialValues={asset} />
    </DialogContent>
  </Dialog>
  );
};

EditAssetDialog.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  asset: PropTypes.shape({
    owners: PropTypes.arrayOf(PropTypes.object).isRequired,
    projects: PropTypes.arrayOf(PropTypes.object).isRequired,
    visible: PropTypes.string.isRequired,
  }),
  formComponent: PropTypes.elementType.isRequired,
};

export default EditAssetDialog;
