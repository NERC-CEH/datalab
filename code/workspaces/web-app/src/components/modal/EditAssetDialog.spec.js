import React from 'react';
import { shallow } from 'enzyme';
import EditAssetDialog from './EditAssetDialog';

describe('EditAssetDialog', () => {
  const mockAsset = {
    name: 'name',
    version: 'version',
    fileLocation: 'fileLocation',
    owners: [],
    projects: [],
    visible: 'PUBLIC',
  };

  const shallowRender = ({
    title = 'Title',
    onSubmit = jest.fn().mockName('onSubmit'),
    onCancel = jest.fn().mockName('onCancel'),
    asset = mockAsset,
    formComponent,
  }) => shallow(
    <EditAssetDialog {...{ title, onSubmit, onCancel, asset, formComponent }} />,
  );

  it('renders correctly passing values to provided from component', () => {
    const FormComponent = () => <div />;
    expect(shallowRender({ formComponent: FormComponent })).toMatchSnapshot();
  });
});
