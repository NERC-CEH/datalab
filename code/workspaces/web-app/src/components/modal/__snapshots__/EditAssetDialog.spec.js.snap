// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAssetDialog renders correctly passing values to provided from component 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="mui-1"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd css-rx25b2-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-xq3hxg-MuiTypography-root-MuiDialogTitle-root"
          id="mui-1"
        >
          Edit Asset
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 makeStyles-root-1 css-k8n68h-MuiTypography-root"
          >
            name:version (fileLocation)
          </h5>
          <div>
            Form: {"initialValues":{"name":"name","version":"version","fileLocation":"fileLocation","owners":[],"projects":[],"visible":"PUBLIC"},"editPermissions":{}}
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
